warp-tls/Network/Wai/Handler/WarpTLS.hs

```diff
@@ -71,17 +73,52 @@ import System.IO.Error (isEOFError, ioeGetErrorType)
 
 ----------------------------------------------------------------
 
--- | Settings for WarpTLS.
-data TLSSettings = TLSSettings {
+-- | Determines where to load the certificate, chain 
+-- certificates, and key from.
+data CertSettings 
+  = CertFromFile {
     certFile :: FilePath
```

Comment:
I prefer not putting record names on sum types, since it creates partial functions and accessors.

Reply:
No problem, I can remove them.

Reply:
One slight problem with removing the fields is that `certFile` and `keyFile` are currently exported by the module, so removing them or changing their types would not be backwards compatible.

Reply:
Leaving it as-is is a problem anyway, since (1) it changes their types already from `TLSSettings -> val` to `CertSettings -> val` and (2) it makes them partial. IMO, it's worth a breaking change to hide these away, since they don't behave the way an outside user would really expect.